[17.0][MIG] stock_picking_partner_note

astirpe opened the pull request.

santostelmo committed:
[ADD] stock_picking_partner_note

Commit:
[UPD] Update stock_picking_partner_note.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_picking_partner_note
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_picking_partner_note/it/

JuMiSanAr committed:
stock_picking_partner_note: only use active notes

OCA-git-bot committed:
[BOT] post-merge updates

santostelmo committed:
[IMP] stock_picking_partner_note check note already in use

Commit:
[UPD] Update stock_picking_partner_note.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_picking_partner_note
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_picking_partner_note/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (28 of 28 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_picking_partner_note
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_picking_partner_note/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (28 of 28 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_picking_partner_note
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_picking_partner_note/it/

astirpe committed:
[MIG] stock_picking_partner_note: pre-commit

astirpe committed:
[MIG] stock_picking_partner_note: migrate to V17

scigghia reviewed (approved):
LGTM

StefanRijnhart commented on stock_picking_partner_note/models/res_config_settings.py:
```diff
@@ -0,0 +1,15 @@
+    check_note_already_in_use = fields.Boolean(
+        related="company_id.check_note_already_in_use",
+        readonly=False,
+        help="That must be activated if you want to prevent the update or deletion "
+        "of a note that is already in use by multiple contacts.",
```
```suggestion
        help="Prevent the update or deletion of a note that is already in use by "
        "more than one contact.",
```

StefanRijnhart commented on stock_picking_partner_note/models/stock_picking_note.py:
```diff
@@ -0,0 +1,39 @@
+            if len(partners) > 1:
+                raise UserError(
+                    _(
+                        "You cannot update or delete a note that linked to multiple"
```
```suggestion
                        "You cannot update or delete a note that is linked to multiple"
```

StefanRijnhart commented on stock_picking_partner_note/readme/CONFIGURE.md:
```diff
@@ -0,0 +1,9 @@
+  types will be shown in picking operations notes.
+- On contacts add picking notes.
+
+You can prevent users ot update or delete notes alredy in use by
```
```suggestion
You can prevent users from updating or deleting notes already in use by
```

StefanRijnhart commented on stock_picking_partner_note/readme/DESCRIPTION.md:
```diff
@@ -0,0 +1,3 @@
+Allow to record a message for the person in charge of order preparation
+at the level of the customer, then have it as a note on the picking
+transfer.
```
```suggestion
transfer. This reuses the existing note field on the picking and will make it read-only.
```

StefanRijnhart commented on stock_picking_partner_note/models/stock_picking.py:
```diff
@@ -0,0 +1,26 @@
+                and n.note_type_id in ptnts
+            )
+            picking_notes = [
+                note.name.strip()
```
Do you want to apply odoo.tools.html_escape to the names, given that we're filling a html field with text content?

astirpe replied:
@StefanRijnhart sorry for the delay, I missed this code review. Thank you BTW! I applied html_escape, would you check?

StefanRijnhart reviewed (commented):
Looking good, just some cosmetic suggestions to consider.

rousseldenis commented:
/ocabot migration stock_picking_partner_note

astirpe committed:
[MIG] stock_picking_partner_note: Improving help messages and readme

Update stock_picking_partner_note/models/res_config_settings.py
Update stock_picking_partner_note/models/stock_picking_note.py
Update stock_picking_partner_note/readme/CONFIGURE.md
Update stock_picking_partner_note/readme/DESCRIPTION.md

Co-authored-by: Stefan Rijnhart (Opener) <stefan@opener.amsterdam>

astirpe committed:
[MIG] stock_picking_partner_note: make use of html_escape in picking note

rousseldenis commented:
@astirpe Thanks for this.

Could you improve your commit messages (you can keep separate commits) :

![image](https://github.com/user-attachments/assets/421e8831-3acc-4f39-a928-1e51aa5a3a8c)


To reflect the real change (e.g.: `[IMP] stock_picking_partner_note: Improving help messages` or ...)

Thanks

StefanRijnhart reviewed (approved):
Thanks for taking my suggestions!

astirpe commented:
@rousseldenis done, thanks!

StefanRijnhart commented:
/ocabot merge nobump
